java/org/contikios/cooja/motes/AbstractEmulatedMote.java

```diff
@@ -31,11 +31,12 @@
 import org.contikios.cooja.Mote;
 import org.contikios.cooja.MoteType;
 import org.contikios.cooja.Simulation;
+import org.contikios.cooja.mote.memory.MemoryInterface;
 import org.contikios.cooja.plugins.BufferListener;
 import org.contikios.cooja.plugins.TimeLine;
 
-public abstract class AbstractEmulatedMote extends AbstractWakeupMote implements Mote {
-  public AbstractEmulatedMote(MoteType moteType, Simulation sim) {
+public abstract class AbstractEmulatedMote<T extends MoteType,M extends MemoryInterface> extends AbstractWakeupMote<T,M> implements Mote {
```

Comment:
Spaces after the commas like in AbstractWakeupMote?

Reply:
Updated. Thanks!